frontend/src/components/RegenerateRecoveryCodeDialog.vue

```diff
@@ -0,0 +1,185 @@
+            </DialogPanel>
+          </TransitionChild>
+        </div>
+      </div>
+    </Dialog>
+  </TransitionRoot>
+</template>
+
+<script setup lang="ts">
+import { Dialog, DialogOverlay, DialogPanel, DialogTitle, TransitionChild, TransitionRoot } from '@headlessui/vue';
+import { ClipboardIcon } from '@heroicons/vue/20/solid';
+import { ArrowPathIcon, KeyIcon } from '@heroicons/vue/24/outline';
+import { base64 } from 'rfc4648';
+import { computed, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import backend from '../common/backend';
+import { BrowserKeys, UserKeys } from '../common/crypto';
+import { JWEBuilder } from '../common/jwe';
+import { debounce } from '../common/util';
+
+enum State {
+  ConfirmRegenerateRecoveryCode,
+  SaveRecoveryCode
+}
+
+const { t } = useI18n({ useScope: 'global' });
+
+const onRegenerateError = ref<Error | null>();
+
+const state = ref(State.ConfirmRegenerateRecoveryCode);
+const processing = ref(false);
+
+const open = ref(false);
+const copiedRecoveryCode = ref(false);
+const debouncedCopyFinish = debounce(() => copiedRecoveryCode.value = false, 2000);
+
+const props = defineProps<{
+  recoveryCode: string
+}>();
+
+const emit = defineEmits<{
+  (e: 'update:recoveryCode', recoveryCode: string): void 
+  (e: 'close'): void
+}>();
+
+defineExpose({
+  show
+});
+
+const recoveryCode = computed({
+  get: () => props.recoveryCode,
+  set: recoveryCode => emit('update:recoveryCode', recoveryCode),
+});
+
+async function show() {
+  state.value = State.ConfirmRegenerateRecoveryCode;
+  open.value = true;
+}
+
+async function regenerateRecoveryCode() {
+  onRegenerateError.value = null;
+  try {
+    processing.value = true;
+
+    const me = await backend.users.me(true);
+    if (me.publicKey == null || me.setupCode == null) {
+      throw new Error('User not initialized.');
+    }
+    const browserKeys = await BrowserKeys.load(me.id);
+    const browserId = await browserKeys.id();
+    const myDevice = me.devices.find(d => d.id == browserId);
+    if (myDevice == null) {
+      throw new Error('Device not initialized.');
+    }
+    const newCode = crypto.randomUUID();
+    const userKeys = await UserKeys.decryptOnBrowser(myDevice.userPrivateKey, browserKeys.keyPair.privateKey, base64.parse(me.publicKey));
+    me.privateKey = await userKeys.encryptedPrivateKey(newCode);
+    me.setupCode = await JWEBuilder.ecdhEs(userKeys.keyPair.publicKey).encrypt({ setupCode: newCode });
+    await backend.users.putMe(me);
+    recoveryCode.value = newCode;
+
+    state.value = State.SaveRecoveryCode;
+  } catch (error) {
+    console.error('Regenerating recovery code failed.', error);
+    onRegenerateError.value = error instanceof Error ? error : new Error('Unknown Error');
+  } finally {
+    processing.value = false;
+  }
+}
+
+async function copyRecoveryCode() {
+  await navigator.clipboard.writeText(recoveryCode.value);
+  copiedRecoveryCode.value = true;
+  debouncedCopyFinish();
+}
+</script>
```

Comment:
Should we rename the file and replace all occurrences of recovery code with setup code?